packages/application-generic/src/logging/log.decorator.ts

```diff
@@ -0,0 +1,38 @@
+import { DecorateAll } from 'decorate-all';
+import { Logger } from '@nestjs/common';
+import { maskValue } from './masking';
+
+export function log() {
+  return (
+    target: any,
+    propertyKey: string,
+    propertyDescriptor: PropertyDescriptor
+  ) => {
+    const originalMethod = propertyDescriptor.value;
+
+    propertyDescriptor.value = async function (...args: any[]) {
+      // { methodName: "name", args: [{"name": "value"}] }
+      const item = {
+        methodName: `${this.constructor.name}.${propertyKey}`,
+        args: [],
+      };
+
+      const promises = args.map(async (value) => {
+        item.args.push({
+          [value.constructor.name]: maskValue(value.constructor.name, value),
+        });
+      });
+
+      await Promise.all(promises);
```

Comment:
Not entirely sure why all this complexity is needed, I think this could be easily refactored to:

```suggestion
for (const argument of args) {
    item.args.push({
      [value.constructor.name]: maskValue(value.constructor.name, value),
    });
}
```

Or perhaps I'm missing something